### PR TITLE
Fix tray styling

### PR DESCRIPTION
**Pull request by claucambra**
Addresses the issues caused by #5377

Principally:

- Simplifies code
- Returns two-line activity titles to improve readability
- Fixes clipping of avatars
- Moves file details button to first row to conserve space 
- Uses our custom button components for consistency
- Fixes dark mode issues
- Properly aligns activity text when secondary text is not there
- more

After changes:

https://github.com/nextcloud/desktop/assets/70155116/c1f098dd-5332-4aff-a7df-91d430b71a76

Screenshot 2023-05-23 at 21 50 27
Screenshot 2023-05-23 at 21 50 53

Before changes:

https://github.com/nextcloud/desktop/assets/70155116/cde9065b-4f67-4882-b10c-b86c857ad072

Screenshot 2023-05-23 at 21 52 07
Screenshot 2023-05-23 at 21 52 21

### CODE REVIEW

File: src/gui/tray/ActivityItemContent.qml [hunk 1]
```diff
@@ -118,32 +114,26 @@ Item {
...
-                                     (activityTextDateTime.visible ? activityTextDateTime.width + Style.trayHorizontalMargin : 0) -
-                                     (dismissActionButton.visible ? dismissActionButton.width + Style.trayHorizontalMargin : 0)
-                Layout.alignment: Qt.AlignTop | Qt.AlignLeft
+                Layout.fillWidth: true
+                Layout.alignment: Qt.AlignVCenter | Qt.AlignLeft
 
                 elide: Text.ElideRight
                 wrapMode: Text.Wrap
-                maximumLineCount: 1
+                maximumLineCount: 2
```
Comment (mgallien) [inline src/gui/tray/ActivityItemContent.qml:136]: do we often get title that long ?

File: src/gui/main.cpp [hunk 2]
```diff
@@ -74,20 +74,9 @@ int main(int argc, char **argv)
     QSurfaceFormat::setDefaultFormat(surfaceFormat);
 
     QQuickWindow::setTextRenderType(QQuickWindow::NativeTextRendering);
-    QQuickStyle::setStyle(QStringLiteral("Fusion"));
```
[the remark this answers is not in the document]
Comment (mgallien) [reply src/gui/main.cpp:77]: I tested your change and I get a badly broken menu when running on plasma desktop

File: src/gui/tray/ActivityItemContent.qml [hunk 3]
```diff
@@ -184,12 +195,13 @@ Item {
 
                 visible: root.showDismissButton && !fileDetailsButton.visible
 
-                icon.source: "image://svgimage-custom-color/clear.svg" + "/" + Style.ncTextColor
+                icon.source: "image://svgimage-custom-color/clear.svg/" + Style.ncTextColor
 
-                flat: true
                 display: Button.IconOnly
-                hoverEnabled: true
```
Comment (mgallien) [inline src/gui/tray/ActivityItemContent.qml:191]: why removing this ?
do not we want hover effect on this button ?

**Review by mgallien — CHANGES_REQUESTED** [review]

**Comment by mgallien** [comment]
@claucambra can you provide screenshot before/after ?

File: src/gui/tray/ActivityItemContent.qml (hunk 3 above)
Comment (claucambra) [reply src/gui/tray/ActivityItemContent.qml:191]: CustomButton already handles this

File: src/gui/tray/ActivityItemContent.qml (hunk 1 above)
Comment (claucambra) [reply src/gui/tray/ActivityItemContent.qml:136]: Most of the time, yes -- a lot of talk messages and file changes within the same folder become unintelligible if we restrict them to oneline

File: theme/Style/Style.qml [hunk 4]
```diff
@@ -84,7 +84,7 @@ QtObject {
     property int addAccountButtonHeight: 50
 
     property int headerButtonIconSize: 32
-    property int dismissButtonSize: 16
+    property int dismissButtonSize: 24
```
[the remark this answers is not in the document]
Comment (claucambra) [reply theme/Style/Style.qml:87]: The touch target on high dpi systems is very small, and this lets us have a bit of extra space to also show the hover effect

File: src/gui/main.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (claucambra) [reply src/gui/main.cpp:77]: Okay, as we spoke about I will remove this commit

We can test the Qt6 fixes PRs

**Comment by claucambra** [comment]
dark mode issues:
Screenshot 2023-05-23 at 21 58 06

**Comment by claucambra** [comment]
> dark mode issues when fusion is enabled: Screenshot 2023-05-23 at 21 58 06

now fixed

**Comment by mgallien** [comment]
@claucambra can you solve the conflicts ?

**Comment by claucambra** [comment]
> @claucambra can you solve the conflicts ?

Yep, now fixed

**Review by mgallien — CHANGES_REQUESTED** [review]
dark mode works fine !!!! super cool
see my inline comments
including relying on implicit magic behavior of our components inheriting existing ones from Qt modules
given the amount of visual changes, I think you should request a design review on this

File: src/gui/ConflictItemFileInfo.qml [hunk 5]
```diff
@@ -37,6 +37,26 @@ Item {
         leftPadding: 0
         spacing: 0
 
+        // TODO: Rather than setting all these palette colours manually,
+        // create a custom style and do it for all components globally
```
[the remark this answers is not in the document]
Comment (claucambra) [reply src/gui/ConflictItemFileInfo.qml:41]: I think with the Qt6 transition we can largely remove these palette definitions. The note is at least helpful when grepping for the places we have had to enforce custom colouring to remove when this happens

**Comment by claucambra** [comment]
> dark mode works fine !!!! super cool

Thanks :)

> see my inline comments including relying on implicit magic behavior of our components inheriting existing ones from Qt modules

I am not convinced by the magic behaviour as the purpose of the CustomButton is to abstract broad visual behaviour so as to not have to repeat it everywhere throughout our code -- that is to say, we shouldn't have to implement things like a hover effect everywhere, which relies on `hoverEnabled: true`

File: src/gui/ConflictItemFileInfo.qml [hunk 6]
```diff
@@ -37,6 +37,26 @@ Item {
...
+            light: Style.lightHover
+            midlight: Style.lightHover
+            mid: Style.ncSecondaryTextColor
+            dark: Style.menuBorder
+            button: Style.menuBorder
+            window: Style.ncTextColor // NOTE: Fusion theme uses darker window colour for the border of the checkbox
+            base: Style.backgroundColor
+            toolTipBase: Style.backgroundColor
+            toolTipText: Style.ncTextColor
+        }
```
[the remark this answers is not in the document]
Comment (claucambra) [reply src/gui/ConflictItemFileInfo.qml]: I have switched all of our windows to be ApplicationWindows and set the palettes here instead

**Comment by mgallien** [comment]
> > dark mode works fine !!!! super cool
> 
> Thanks :)
> 
> > see my inline comments including relying on implicit magic behavior of our components inheriting existing ones from Qt modules
> 
> I am not convinced by the magic behaviour as the purpose of the CustomButton is to abstract broad visual behaviour so as to not have to repeat it everywhere throughout our code -- that is to say, we shouldn't have to implement things like a hover effect everywhere, which relies on `hoverEnabled: true` already being set in `CustomButton`

if what you want to achieve is that by default all instances of `CustomButton` do have hover effect enabled then you can just set a default value for the property inside `CustomButton`
if someone wants to use the component it may decide to change the value of the property and currently that would have no effect looking like a bug (and I would consider this a bug)

**Comment by claucambra** [comment]
> > > dark mode works fine !!!! super cool
> > 
> > 
> > Thanks :)
> > > see my inline comments including relying on implicit magic behavior of our components inheriting existing ones from Qt modules
> > 
> > 
> > I am not convinced by the magic behaviour as the purpose of the CustomButton is to abstract broad visual behaviour so as to not have to repeat it everywhere throughout our code -- that is to say, we shouldn't have to implement things like a hover effect everywhere, which relies on `hoverEnabled: true` already being set in `CustomButton`
> 
> if what you want to achieve is that by default all instances of `CustomButton` do have hover effect enabled then you can just set a default value for the property inside `CustomButton` if someone wants to use the component it may decide to change the value of the property and currently that would have no effect looking like a bug (and I would consider this a bug)

Okay, I declared it explicitly in CustomButton now

File: src/gui/tray/Window.qml [hunk 7]
```diff
@@ -285,11 +304,12 @@ ApplicationWindow {
                         width: (Style.currentAccountButtonWidth - 2)
                         height: Math.min(implicitHeight, maxMenuHeight)
                         closePolicy: Menu.CloseOnPressOutsideParent | Menu.CloseOnEscape
-                        palette: Style.palette
+
```
[the remark this answers is not in the document]
Comment (claucambra) [reply src/gui/tray/Window.qml:307]: Removed

**Review by mgallien — APPROVED** [review]

**Pull request merged**